### PR TITLE
Add `overlapping-except` example

### PR DESCRIPTION
**Pull request by harupy**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Refs #5953

Closes #XXXX

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I think an example using a builtin exception would be easier to understand. It would also show the this do not need to be the exact same exception, only a children. Maybe ``ArithmeticError``, and ``FloatingPointError`` (Arithmetic error catch  FloatingPointError, OverflowError and ZeroDivisionError) ? But there's also a lot of possible example for OsError in in https://docs.python.org/3/library/exceptions.html#exception-hierarchy.

I think we could add this link in details.rst too so the hierarchy for builtin is easy to find .

**Comment by harupy** [comment]
@Pierre-Sassoulas Sounds good. I'll fix the code.

File: doc/data/messages/o/overlapping-except/bad.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+try:
+    pass
+except (ArithmeticError, FloatingPointError):  # [overlapping-except]
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/o/overlapping-except/bad.py:4]:
```suggestion
def divide_x_by_y(x: float, y: float):
    try:
        print(x / y)
    except (ArithmeticError, FloatingPointError) as e:  # [overlapping-except]
        print(f"There was an issue : {e}")
```

File: doc/data/messages/o/overlapping-except/good.py [hunk 2]
```diff
@@ -1 +1,11 @@
...
+    pass
+except ArithmeticError:
+    pass
+
+# or
+
+try:
+    pass
+except FloatingPointError:
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/o/overlapping-except/good.py:11]:
```suggestion
def divide_x_by_y(x: float, y: float):
    try:
        print(x / y)
    except FloatingPointError as e:
        print(f"There was a FloatingPointError: {e}")
    except ArithmeticError as e:
        # FloatingPointError  were already caught at this point
        print(f"There was an OverflowError or a ZeroDivisionError: {e}")

Or:

def divide_x_by_y(x: float, y: float):
    try:
        print(x / y)
    except ArithmeticError as e:
        print(f"There was an OverflowError, a ZeroDivisionError or a FloatingPointError : {e}")
```

File: doc/data/messages/o/overlapping-except/good.py [hunk 3]
```diff
@@ -1 +1,16 @@
...
+        # FloatingPointError  were already caught at this point
+        print(f"There was an OverflowError or a ZeroDivisionError: {e}")
+
+Or:
+
+def divide_x_by_y(x: float, y: float):
+    try:
+        print(x / y)
+    except ArithmeticError as e:
+        print(f"There was an OverflowError, a ZeroDivisionError or a FloatingPointError : {e}")
```
Comment (harupy) [inline doc/data/messages/o/overlapping-except/good.py:16]:
```suggestion
        print(f"There was an OverflowError, a ZeroDivisionError or a FloatingPointError: {e}")
```

File: doc/data/messages/o/overlapping-except/bad.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+def divide_x_by_y(x: float, y: float):
+    try:
+        print(x / y)
+    except (ArithmeticError, FloatingPointError) as e:  # [overlapping-except]
+        print(f"There was an issue : {e}")
```
Comment (harupy) [inline doc/data/messages/o/overlapping-except/bad.py:5]:
```suggestion
        print(f"There was an issue: {e}")
```

File: doc/data/messages/o/overlapping-except/good.py [hunk 5]
```diff
@@ -1 +1,16 @@
...
+def divide_x_by_y(x: float, y: float):
+    try:
+        print(x / y)
+    except FloatingPointError as e:
+        print(f"There was a FloatingPointError: {e}")
+    except ArithmeticError as e:
+        # FloatingPointError  were already caught at this point
+        print(f"There was an OverflowError or a ZeroDivisionError: {e}")
+
+Or:
```
Comment (harupy) [inline doc/data/messages/o/overlapping-except/good.py:10]:
```suggestion
# Or:
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**